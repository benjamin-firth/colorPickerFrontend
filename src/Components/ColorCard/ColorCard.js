import React, { useState } from 'react';
import './ColorCard.scss';
import PropTypes from 'prop-types';

function ColorCard({ id }) {
  let randomColor = Math.floor(Math.random()*16777215).toString(16);
  //set global state to randomColor
  const [ locked, switchLock ] = useState();

  return (
    <div className={`color-card color-card${id}`} style={{
      backgroundColor: `#${randomColor}`
    }}>
      <p>lock</p>
      <p>#{randomColor.toUpperCase()}</p>
    </div>
  );
}

export default ColorCard;
