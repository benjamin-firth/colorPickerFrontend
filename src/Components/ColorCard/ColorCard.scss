.color-card {
    border: 2px solid black;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-end;
    height: 15vw;
    width: 15vw;
    margin: .5vw;
    p {
      margin: 0;
    }
    button {
      background-color: transparent;
      border: none;
    }
};

.color-card1 {
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
}

.color-card2 {
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
}

.color-card3 {
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
}

.color-card4 {
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
}

.color-card5 {
  border-radius: 200px 5px 200px 5px/5px 255px 5px 215px;
}
