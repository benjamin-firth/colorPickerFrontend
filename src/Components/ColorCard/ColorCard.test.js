import React from 'react';
import { render } from '@testing-library/react';
import ColorCard from './ColorCard';
import { shallow } from 'enzyme';

describe('', () => {
  it('should', () => {

  })
});
