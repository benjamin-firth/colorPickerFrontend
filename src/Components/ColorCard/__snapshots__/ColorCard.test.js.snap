// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match the snapshot 1`] = `
<div
  className="color-card color-card2"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <button
    className="lock-button"
    onClick={[Function]}
  >
    lock
  </button>
  <p>
    #
    FFFFFF
  </p>
</div>
`;
